### DISCUSSION TITLE
Build error in windows platform

### DISCUSSION THREAD

**Discussion by psudeepth** [AcademySoftwareFoundation/OpenRV#101] [Build Issues]
While building I am getting the below error

Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.19044.
The CXX compiler identification is MSVC 19.29.30148.0
The C compiler identification is MSVC 19.29.30148.0
Detecting CXX compiler ABI info
Detecting CXX compiler ABI info - done
Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
Detecting CXX compile features
Detecting CXX compile features - done
Detecting C compiler ABI info
Detecting C compiler ABI info - done
Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
Detecting C compile features
Detecting C compile features - done
Build type: Release
PROJECT DIR C:/OpenRV
Building RV for Microsoft Windows
Using build branch: main
Using build hash: 2b799c585
RV_STAGE_ROOT_DIR: C:/OpenRV/Build/stage/app
RV_STAGE_BIN_DIR: C:/OpenRV/Build/stage/app/bin
RV_STAGE_LIB_DIR: C:/OpenRV/Build/stage/app/lib
RV_STAGE_INCLUDE_DIR: C:/OpenRV/Build/stage/app/include
RV_STAGE_SRC_DIR: C:/OpenRV/Build/stage/app/src
RV_STAGE_RESOURCES_DIR: C:/OpenRV/Build/stage/app/resources
RV_STAGE_PLUGINS_DIR: C:/OpenRV/Build/stage/app/PlugIns
RV_STAGE_PLUGINS_CONFIGFILES_DIR: C:/OpenRV/Build/stage/app/PlugIns/ConfigFiles
RV_STAGE_PLUGINS_IMAGEFORMATS_DIR: C:/OpenRV/Build/stage/app/PlugIns/ImageFormats
RV_STAGE_PLUGINS_MOVIEFORMATS_DIR: C:/OpenRV/Build/stage/app/PlugIns/MovieFormats
RV_STAGE_PLUGINS_MU_DIR: C:/OpenRV/Build/stage/app/PlugIns/Mu
RV_STAGE_PLUGINS_NODES_DIR: C:/OpenRV/Build/stage/app/PlugIns/Nodes
RV_STAGE_PLUGINS_OIIO_DIR: C:/OpenRV/Build/stage/app/PlugIns/OIIO
RV_STAGE_PLUGINS_PACKAGES_DIR: C:/OpenRV/Build/stage/app/PlugIns/Packages
RV_STAGE_PLUGINS_PROFILES_DIR: C:/OpenRV/Build/stage/app/PlugIns/Profiles
RV_STAGE_PLUGINS_PYTHON_DIR: C:/OpenRV/Build/stage/app/PlugIns/Python
RV_STAGE_PLUGINS_QT_DIR: C:/OpenRV/Build/stage/app/PlugIns/Qt
RV_STAGE_PLUGINS_SUPPORTFILES_DIR: C:/OpenRV/Build/stage/app/PlugIns/SupportFiles
Updating submodules
Copying Qt into C:/OpenRV/Build/stage/app
Copying Qt translations files ...
Copying Qt resources files ...
Copying Qt into C:/OpenRV/Build/stage/app -- DONE
Populating rv_deps_opentimelineio
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.19044.

-- Configuring done (3.6s)

-- Generating done (0.1s)

-- Build files have been written to: C:/OpenRV/Build/_deps/rv_deps_opentimelineio-subbuild

Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System

  Creating directories for 'rv_deps_opentimelineio-populate'

  Building Custom Rule C:/OpenRV/Build/_deps/rv_deps_opentimelineio-subbuild/CMakeLists.txt

  Performing download step (git clone) for 'rv_deps_opentimelineio-populate'

  Cloning into 'rv_deps_opentimelineio-src'...

  HEAD is now at 3703ac7 Release v0.15.0 changes (#1403)

  Submodule 'src/deps/Imath' (https://github.com/AcademySoftwareFoundation/Imath) registered for path 'src/deps/Imath'

  Submodule 'src/deps/any' (https://github.com/thelink2012/any.git) registered for path 'src/deps/any'
  Submodule 'src/deps/optional-lite' (https://github.com/martinmoene/optional-lite) registered for path 'src/deps/optional-lite'
  Submodule 'src/deps/pybind11' (https://github.com/pybind/pybind11.git) registered for path 'src/deps/pybind11'
  Submodule 'src/deps/rapidjson' (https://github.com/Tencent/rapidjson.git) registered for path 'src/deps/rapidjson'

  Cloning into 'C:/OpenRV/Build/_deps/rv_deps_opentimelineio-src/src/deps/Imath'...

  Cloning into 'C:/OpenRV/Build/_deps/rv_deps_opentimelineio-src/src/deps/any'...

  Cloning into 'C:/OpenRV/Build/_deps/rv_deps_opentimelineio-src/src/deps/optional-lite'...

  Cloning into 'C:/OpenRV/Build/_deps/rv_deps_opentimelineio-src/src/deps/pybind11'...

  Cloning into 'C:/OpenRV/Build/_deps/rv_deps_opentimelineio-src/src/deps/rapidjson'...

  Submodule path 'src/deps/Imath': checked out '3ad5d4da02795dbf37b44c5bbd2f919a7c01a19a'

  Submodule path 'src/deps/any': checked out 'bfc77f2e4be6e9a093dd645ef3f1aa30620c9205'

  Submodule path 'src/deps/optional-lite': checked out '5e4b7760aac85ea73a5b9b91c726fc36cbd4149b'

  Submodule path 'src/deps/pybind11': checked out '914c06fb252b6cc3727d0eedab6736e88a3fcb01'

  Submodule path 'src/deps/rapidjson': checked out '06d58b9e848c650114556a23294d0b6440078c61'

  Submodule 'thirdparty/gtest' (https://github.com/google/googletest.git) registered for path 'src/deps/rapidjson/thirdparty/gtest'

  Cloning into 'C:/OpenRV/Build/_deps/rv_deps_opentimelineio-src/src/deps/rapidjson/thirdparty/gtest'...

  Submodule path 'src/deps/rapidjson/thirdparty/gtest': checked out 'ba96d0b1161f540656efdaed035b3c062b60e006'

  Performing update step for 'rv_deps_opentimelineio-populate'

  No patch step for 'rv_deps_opentimelineio-populate'

  No configure step for 'rv_deps_opentimelineio-populate'

  No build step for 'rv_deps_opentimelineio-populate'

  No install step for 'rv_deps_opentimelineio-populate'

  No test step for 'rv_deps_opentimelineio-populate'

  Completed 'rv_deps_opentimelineio-populate'

  Building Custom Rule C:/OpenRV/Build/_deps/rv_deps_opentimelineio-subbuild/CMakeLists.txt

Populating rv_deps_pyside2
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.19044.

-- Configuring done (1.4s)

-- Generating done (0.1s)

-- Build files have been written to: C:/OpenRV/Build/_deps/rv_deps_pyside2-subbuild

Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System

  Creating directories for 'rv_deps_pyside2-populate'

  Building Custom Rule C:/OpenRV/Build/_deps/rv_deps_pyside2-subbuild/CMakeLists.txt

  Performing download step (git clone) for 'rv_deps_pyside2-populate'

  Cloning into 'rv_deps_pyside2-src'...

  HEAD is now at 9282e03de Enable 5.15.2.1 fix release

  Submodule 'sources/pyside2-tools' (git://code.qt.io/pyside/pyside-tools.git) registered for path 'sources/pyside2-tools'

  Cloning into 'C:/OpenRV/Build/_deps/rv_deps_pyside2-src/sources/pyside2-tools'...

  Submodule path 'sources/pyside2-tools': checked out 'a8448837204faee0b457d1e2d4cbf574a2811114'

  Performing update step for 'rv_deps_pyside2-populate'

  No patch step for 'rv_deps_pyside2-populate'

  No configure step for 'rv_deps_pyside2-populate'

  No build step for 'rv_deps_pyside2-populate'

  No install step for 'rv_deps_pyside2-populate'

  No test step for 'rv_deps_pyside2-populate'

  Completed 'rv_deps_pyside2-populate'

  Building Custom Rule C:/OpenRV/Build/_deps/rv_deps_pyside2-subbuild/CMakeLists.txt

Using Boost:        1.76.0
Using Dav1d:        1.0.0
Using FFMPEG:       n4.4.3
Using atomic_ops:   7.7.0
Using Imath:        3.1.5
Using OpenEXR:      3.1.5
Using Python3:      3.9.15
Using PySide2:      5.15.2.1
Using Qt5:          5.15.2
Using zlib:         1.2.13

CMake Error at cmake/macros/rv_stage.cmake:405 (EXECUTE_PROCESS):
  EXECUTE_PROCESS error getting child return code: The system cannot find the
  file specified
Call Stack (most recent call first):
  src/plugins/rv-packages/additional_nodes/CMakeLists.txt:11 (RV_STAGE)

Configuring incomplete, errors occurred!

I Am using cmake-GUI for building With command line also getting the same error. I have configured as per the documentation and installed all the pre-requrisites mentioned.

 Any help will be appreciated.

**Comment by rogernelson** [comment]
This is just a guess, but pyside has some really long paths in it and windows (by default) has a 260 character limit on file paths.  What is the root directory you are building in?  If it's something really long perhaps try to move it as close to the root as possible to shorten the length of the paths.

**Comment by psudeepth** [comment]
Hello rogernelson ,

I Have read in the the thread about the issue so I have kept the root  in C:\OpenRV.

Have also figured out  where its going wrong. i found that 
OpenRV/tree/main/cmake/macros/rv_stage.cmake line: 405 in execute_process its showing file not found but while printing the path variable its printing properly. Even its having a check if the file exists or not (line: 401) and exception is not caught. it is looking for this file OpenRV/tree/main/src/plugins/rv-packages/additional_nodes/PACKAGE and its available in the folder.

**Reply by rogernelson** [reply]
I see the command it's trying to run on that line is:
```
bash -c "cat ${_package_file} | grep version: | grep --only-matching -e '[0-9.]*'"
```

Perhaps its not the package file that it's complaining about, maybe its `bash` or `cat` or `grep` or something? Could you try just running that command your shell and see if it works? Substituting in the appropriate path, it should be something like:

```
bash -c "cat src/plugins/rv-packages/additional_nodes/PACKAGE | grep version: | grep --only-matching -e '[0-9.]*'"
```

**Reply by psudeepth** [reply]
I am building in windows 10 using CMAKE and Visual Studio 2019

**Reply by rogernelson** [reply]
You are building in a mingw64 shell though, right? What I mean is to execute the command inside of that shell.

**Reply by psudeepth** [reply]
Ran the command 
bash -c "cat C:/OpenRV/src/plugins/rv-packages/additional_nodes/PACKAGE | grep version: | grep --only-matching -e '[0-9.]*'"

Result: 
pasud@DESKTOP-RJISE5K MINGW64 ~
# bash -c "cat C:/OpenRV/src/plugins/rv-packages/additional_nodes/PACKAGE | grep version: | grep --only-matching -e '[0-9.]*'"
1.2

The expected result is found. I am using current generator as Visual Studio 16 2019 do I need to change it to MSYS2 MINGW64?

**Reply by rogernelson** [reply]
Since the command works, I suspect then that it is something else that is failing.

Visual Studio 2019 is what you want on the cmake command line.  QtWebEngine won't build with anything else, so it's required. We're only really using msys so we can use standard posix commands like grep from cmake (as you see in the command you ran above). The build process is not using any of the msys/mingw64 compilers or build toolchains.  

You will also need Visual Studio 2017 installed though, since pyside will only build with that. Do you have it on your machine? If it's missing that could also explain the error.

**Reply by psudeepth** [reply]
I have both Visual Studio 2019 and 2017 installed. Both Community edition and all the requirements mentioned in the Building Open RV on Windows. I am using VMWare Fusion on Mac and installed windows as a VM. Will that be any kind of issue?

**Comment by ddesmond** [comment]
check this guide-tips
https://github.com/AcademySoftwareFoundation/OpenRV/discussions/91

**Comment by psudeepth** [comment]
Totally sorry. I am new to build process with CMAKE. Previously I was trying to build with CMAKE-GUI. That is the reason I am getting the above error. I managed to build the source properly but getting many warnings and 1 Error. After the build process is done rv.exe is missing. I followed the process mentioned in #91. The process I followed is  mentioned below and the build log is also attached.

Process: https://github.com/psudeepth/OpenRVBuild/blob/master/Windows_Build_Process.txt
Build Log: https://github.com/psudeepth/OpenRVBuild/blob/master/full_build.log

Any help is much appriciated

**Comment by scottmcdonnell** [comment]
I was getting the same error and it was for bash not being found. 

bash.exe path is correctly set in cmd.exe and msys2 console - both show `where bash` correctly - but in the context of the script running bash is not found.   

Try hardcoding the full path to bash such as "C:/msys2/usr/bin/bash.exe" as a test and see if it moves past that line.

`COMMAND "C:/msys2/usr/bin/bash.exe" -c "cat ${_package_file} | grep version: | grep --only-matching -e '[0-9.]*'"`
